src/GlobalPayments.Api/Terminals/INGENICO/IngenicoController.cs

```diff
@@ -0,0 +1,293 @@
+﻿using GlobalPayments.Api.Builders;
+using GlobalPayments.Api.Entities;
+using GlobalPayments.Api.Terminals.Abstractions;
+using GlobalPayments.Api.Terminals.Builders;
+using GlobalPayments.Api.Terminals.Ingenico.Responses;
+using GlobalPayments.Api.Utils;
+using System;
+using System.Text;
+
+namespace GlobalPayments.Api.Terminals.Ingenico {
+    public class IngenicoController : DeviceController {
+        IDeviceInterface _device;
+
+        public IngenicoController(ITerminalConfiguration settings) : base(settings) {
+        }
+
+        internal override IDeviceInterface ConfigureInterface() {
+            if (_device == null) {
+                _device = new IngenicoInterface(this);
+            }
+            return _device;
+        }
+
+        internal override IDeviceCommInterface ConfigureConnector() {
+            switch (_settings.ConnectionMode) {
+                case ConnectionModes.SERIAL:
+                    return new IngenicoSerialInterface(_settings);
+                case ConnectionModes.TCP_IP:
```

Comment:
This basically says if you pick any of these modes you get TCP and this is misleading. If the two options are Serial and TCP_Server then all the rest should fall to default and a not implemented exception should be thrown.

Reply:
Though now that I think about it, it should probably a different exception given that we should not be throwing anything which is not wrapped in an ApiException

Reply:
So right now, I'll just stick with the existing thrown exception. Just let me know when we'll changed it.